lib/rmt/cli/products.rb

```diff
@@ -9,9 +9,15 @@ class ProductNotFoundException < StandardError
   option :all, aliases: '-a', type: :boolean, desc: _('List all products, including ones which are not marked to be mirrored')
   option :release_stage, aliases: '-r', type: :string, desc: 'beta, released'
   option :csv, type: :boolean, desc: _('Output data in CSV format')
+  option :name, type: :string, default: '', desc: _('Product name (e.g.: Basesystem, SLES)')
+  option :version, type: :string, default: '', desc: _('Product version (e.g.: 15, 15.1, \'12 SP4\')')
+  option :arch, type: :string, default: '', desc: _('Product architecture (e.g.: x86_64, aarch64)')
   def list
     products = (options.all ? Product.all : Product.mirrored).order(:name, :version, :arch)
     products = products.with_release_stage(options[:release_stage])
+    products = products.with_name_filter(options[:name])
+    products = products.with_version_filter(options[:version])
+    products = products.with_arch_filter(options[:arch])
     decorator = ::RMT::CLI::Decorators::ProductDecorator.new(products)
 
     if products.empty?
```

Comment:
You'll have to do a small adjustment on the if/else statement below. When the user has provided any of your new filters, and there are no results, it shouldn't ask the user to run 'rmt-cli sync' as it currently does, but use the other text "No matching products found in database."

Reply:
Good catch, done.